v2/src/datadog.ts

```diff
@@ -0,0 +1,109 @@
+import * as logs from "aws-cdk-lib/aws-logs";
+import { Construct } from "constructs";
+import log from "loglevel";
+import { Transport } from "./common/transport";
+import {
+  applyLayers,
+  redirectHandlers,
+  addForwarder,
+  addForwarderToLogGroups,
+  applyEnvVariables,
+  validateProps,
+  TagKeys,
+  DatadogProps,
+  DatadogStrictProps,
+  handleSettingPropDefaults,
+} from "./index";
+
+const versionJson = require("../version.json");
+
+export class Datadog extends Construct {
+  scope: Construct;
+  props: DatadogProps;
+  transport: Transport;
+  constructor(scope: Construct, id: string, props: DatadogProps) {
+    if (process.env.DD_CONSTRUCT_DEBUG_LOGS?.toLowerCase() == "true") {
+      log.setLevel("debug");
+    }
+    super(scope, id);
+    this.scope = scope;
+    this.props = props;
+    validateProps(this.props);
+    this.transport = new Transport(
+      this.props.flushMetricsToLogs,
+      this.props.site,
+      this.props.apiKey,
+      this.props.apiKeySecretArn,
+      this.props.apiKmsKey,
+      this.props.extensionLayerVersion,
+    );
+  }
+
+  public addLambdaFunctions(
+    lambdaFunctions: (lambda.Function | lambdaNodejs.NodejsFunction | lambdaPython.PythonFunction)[],
+  ) {
+    // baseProps contains all properties set by the user, with default values for properties
+    // defined in DefaultDatadogProps (if not set by user)
+    const baseProps: DatadogStrictProps = handleSettingPropDefaults(this.props);
+
+    if (this.props !== undefined && lambdaFunctions.length > 0) {
+      const region = `${lambdaFunctions[0].env.region}`;
+      log.debug(`Using region: ${region}`);
+      applyLayers(
+        this.scope,
+        region,
+        lambdaFunctions,
+        this.props.pythonLayerVersion,
+        this.props.nodeLayerVersion,
+        this.props.extensionLayerVersion,
+      );
+      redirectHandlers(lambdaFunctions, baseProps.addLayers);
+
+      if (this.props.forwarderArn !== undefined) {
+        if (this.props.extensionLayerVersion !== undefined) {
+          log.debug(`Skipping adding subscriptions to the lambda log groups since the extension is enabled`);
+        } else {
+          log.debug(`Adding log subscriptions using provided Forwarder ARN: ${this.props.forwarderArn}`);
+          addForwarder(this.scope, lambdaFunctions, this.props.forwarderArn);
+        }
+      } else {
+        log.debug("Forwarder ARN not provided, no log group subscriptions will be added");
+      }
+
+      addCdkConstructVersionTag(lambdaFunctions);
+
+      applyEnvVariables(lambdaFunctions, baseProps);
+
+      this.transport.applyEnvVars(lambdaFunctions);
+    }
+  }
+
+  public addForwarderToNonLambdaLogGroups(logGroups: logs.ILogGroup[]) {
+    if (this.props.forwarderArn !== undefined) {
+      addForwarderToLogGroups(this.scope, logGroups, this.props.forwarderArn);
+    } else {
+      log.debug("Forwarder ARN not provided, no non lambda log group subscriptions will be added");
+    }
+  }
```

Comment:
I can move this function into common, though I don't think it's worth it due to the size of interface required:

```
export interface ILogGroup {
  readonly logGroupArn: string;
  readonly logGroupName: string;
  addStream(id: string, props?: any): any;
  addSubscriptionFilter(id: string, props: any): any;
  addMetricFilter(id: string, props: any): any;
  extractMetric(jsonField: string, metricNamespace: string, metricName: string): any;
  grantWrite(grantee: any): any;
  grant(grantee: any, ...actions: string[]): any;
  logGroupPhysicalName(): string;
  addToResourcePolicy(statement: any): any;
  stack: any;
  env: any; 
  node: any;
}
```

Reply:
we've decided it's not worth it for now